corehq/apps/data_dictionary/util.py

```diff
@@ -196,7 +204,7 @@ def get_data_dict_props_by_case_type(domain):
     return {
         case_type: {prop.name for prop in props} for case_type, props in groupby(
             CaseProperty.objects
-            .filter(case_type__domain=domain, deprecated=False)
+            .filter(case_type__domain=domain)
```

Comment:
What's the purpose of this change? This function is called in the [ParentCasePropertyBuilder](https://github.com/dimagi/commcare-hq/blob/06f4aa6744babf691ed5b0a42049cb6856378d3b/corehq/apps/app_manager/app_schemas/case_properties.py#L373), which makes me think it might make deprecated properties show up in places beyond the case importer, such as the "App Properties" window that shows up when you're building a form.

Reply:
Oops I think I forgot to track down if this was used elsewhere. I've updated the PR to include a new param that allows for return of deprecated properties when desired (the f/e here wants to warn when one is typed or suggested, and also to show the valid values and formats even for deprecated ones).